### PR TITLE
feat(minor-voting-verifier)!: use transaction log index instead of block log index to identify EVM events

### PR DESCRIPTION
**Pull request by fish-sammy**
## Todos

- [x] Unit tests
- [ ] Manual tests
- [ ] Documentation
- [ ] Connect epics/issues

### CODE REVIEW

File: ampd/src/evm/verifier.rs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 use axelar_wasm_std::voting::Vote;
+use cast::usize;
```
Comment (cgorenflo) [inline ampd/src/evm/verifier.rs:2]: where is it used?

File: contracts/nexus-gateway/src/nexus.rs [hunk 2]
```diff
@@ -25,6 +25,8 @@ pub struct Message {
 
 impl CustomMsg for Message {}
 
+// it's parsed into u64 instead of u32 (https://github.com/axelarnetwork/axelar-amplifier/blob/bf0b3049c83e540989c7dad1c609c7e2ef6ed2e5/contracts/voting-verifier/src/events.rs#L162)
+// here in order to match the message type defined in the nexus module
```
Comment (cgorenflo) [inline contracts/nexus-gateway/src/nexus.rs:29]:
```suggestion
// here in order to match the message type defined in the nexus module. Changing nexus to use u32 instead is not worth the effort.
```

File: Cargo.toml [hunk 3]
```diff
@@ -35,6 +35,7 @@ serde_json = "1.0.89"
 schemars = "0.8.10"
 sha3 = { version = "0.10.8", default-features = false, features = [] }
 signature-verifier-api = { version = "^0.1.0", path = "packages/signature-verifier-api" }
+cast = "0.3.0"
```
Comment (cgorenflo) [inline Cargo.toml:38]: do we need both `cast` and `num-traits`?

**Review by cgorenflo — APPROVED** [review]

File: ampd/src/evm/verifier.rs (hunk 1 above)
Comment (fish-sammy) [reply ampd/src/evm/verifier.rs:2]: https://github.com/axelarnetwork/axelar-amplifier/blob/34d4a541897c1a53a6f54c9b2f2be4487d619ebb/ampd/src/evm/verifier.rs#L75

File: Cargo.toml (hunk 3 above)
Comment (fish-sammy) [reply Cargo.toml:38]: oh I didn't know we had `num-traits`. Let me check.

**Pull request merged**